### PR TITLE
Terra upgrade

### PR DESCRIPTION
**Pull request by sen-p**
### Details and comments
I created .py files from the examples provided for the dm-simulator to run tests. Please ignore those changes

### CODE REVIEW

File: qiskit/execute_function.py [hunk 1]
```diff
@@ -324,8 +324,7 @@ def execute(
 
     if isinstance(backend, BaseBackend):
         # assembling the circuits into a qobj to be run on the backend
-        if shots is None:
```
Comment (SamarthH) [inline qiskit/execute_function.py:327]: I do not think removing this is a good idea. What one should rather do is that one should add another check of if the backend is DM Simulator, and only in that case should one keep shots None.

File: qiskit/providers/basicaer/basicaertools.py [hunk 2]
```diff
@@ -737,7 +737,7 @@ def partition(i_set, num_qubits):
         if mod_ins != []:
             # Bell, Expect and Ensemble measure form a partitiom on their own.
             if mod_ins[0].name=='measure' and getattr(mod_ins[0],'params',None) != None and mod_ins[0].params[0] in ['Bell', 'Expect', 'Ensemble']:
-                partition_list.append(mod_ins)
+                partition_list.append([mod_ins])
```
Comment (SamarthH) [inline qiskit/providers/basicaer/basicaertools.py:740]: Why was this change made?

File: qiskit/execute_function.py (hunk 1 above)
Comment (obliviateandsurrender) [inline qiskit/execute_function.py:327]: Yes, I think we already specify the maximum shots limit for `dm_simulator` as 1.

**Comment by SamarthH** [comment]
@obliviateandsurrender We are just in the process of updating the backend to BackendV1. This is just a stepping stone. It will be cleaned up soon.

File: qiskit/execute_function.py (hunk 1 above)
Comment (SamarthH) [reply qiskit/execute_function.py:327]: This change was done to preserve the use of BaseBackend. We will shift away from this soon

Comment (biplab37) [reply qiskit/execute_function.py:327]: As far as I remember, even if the number of shots are specified the code in dm_simulator does not use it. But removing the shots will make it not compatible with the other backends. So there is no need to change anything about shots in this file. 
The information about the number of shots goes to the [`self.run_experiment`](https://github.com/indian-institute-of-science-qc/qiskit-aakash/blob/09ac4b66f6f8cea09ac82f17bf8dd38ab6fb983e/qiskit/providers/basicaer/dm_simulator.py#L950) function but the  function inside the `dm_simulator` does not use it as concept of shot is not present in density matrix formalism.

**Pull request merged**